src/examples/IK/exampleIK.cpp

```diff
@@ -141,236 +182,291 @@ const std::vector<std::string> getJointsList()
+    initialJointVelocities.resize(kinDyn->getNrOfDegreesOfFreedom()); // Resize the initial joint velocity vector based on the number of degrees of freedom
+    initialJointPositions.setZero(); // Set initial joint positions to zero
+    gravity << 0.0, 0.0, -9.81; // Assign values to the gravity vector
+    basePose.setIdentity(); // Set the base pose to the identity matrix
+    baseVelocity.setZero(); // Set base velocity to zero
+
+
+    auto paramHandler = std::make_shared<BipedalLocomotion::ParametersHandler::YarpImplementation>(); // Create a parameter handler
+
+    // Set parameters from a configuration file
+    if (!paramHandler->setFromFile(getConfigPath() + "/exampleIK.ini")) // Check if setting parameters from file is successful
     {
-        BiomechanicalAnalysis::log()->error("Cannot configure the parameter handler");
-        return 1;
+        BiomechanicalAnalysis::log()->error("Cannot configure the parameter handler"); // Log error if configuration fails
+        return 1; // Return error code
     }
 
-    matioCpp::File file("/path/to/matlab1.mat");
-
-    matioCpp::Struct ifeel_data = file.read("ifeel_data").asStruct();
-    matioCpp::Struct node12 = ifeel_data("iFeelSuit_vLink_Node_12").asStruct();
-
-    matioCpp::File file2("/path/to/human_data.mat");
-    matioCpp::Struct human_data = file2.read("human_data").asStruct();
-    matioCpp::Struct human_state = human_data("human_state").asStruct();
-    matioCpp::Struct joint_positions = human_state("joint_positions").asStruct();
-    matioCpp::MultiDimensionalArray<double> jointPos_data
-        = joint_positions("data").asMultiDimensionalArray<double>();
-    matioCpp::Struct base_position = human_state("base_position").asStruct();
-    matioCpp::MultiDimensionalArray<double> basePos_data
-        = base_position("data").asMultiDimensionalArray<double>();
-    matioCpp::Struct base_orientation = human_state("base_orientation").asStruct();
-    matioCpp::MultiDimensionalArray<double> baseOrientation_data
-        = base_orientation("data").asMultiDimensionalArray<double>();
-
+    // Load data from Matlab files
+    matioCpp::File file("/path/to/matlab1.mat");// Create a file object to read data from MATLAB file "matlab1.mat"
+    matioCpp::Struct ifeel_data = file.read("ifeel_data").asStruct();// Read "ifeel_data" structure from the MATLAB file and convert it to a structure
+    matioCpp::Struct node12 = ifeel_data("iFeelSuit_vLink_Node_12").asStruct();// Access structure "iFeelSuit_vLink_Node_12" within "ifeel_data" structure and convert it to a structure
+    matioCpp::File file2("/path/to/human_data.mat");    matioCpp::Struct human_data = file2.read("human_data").asStruct(); // Read "human_data" structure from the MATLAB file and convert it to a structure
+    matioCpp::Struct human_data = file2.read("human_data").asStruct(); // Read "human_data" structure from the MATLAB file and convert it to a structure    
```

Comment:
Here the code `matioCpp::Struct human_data = file2.read("human_data").asStruct();` is repeated and is causing the failure of the CI.

Reply:
fixed thanks @davidegorbani